pilot/pkg/networking/core/listener_builder.go

```diff
@@ -307,9 +308,29 @@ func (lb *ListenerBuilder) buildHTTPConnectionManager(httpOpts *httpListenerOpts
 	if httpOpts.http3Only {
 		connectionManager.CodecType = hcm.HttpConnectionManager_HTTP3
 		connectionManager.Http3ProtocolOptions = &core.Http3ProtocolOptions{}
+	} else if lb.push.Mesh.GetDefaultConfig().GetProxyHeaders().GetPreserveHttp1HeaderCase() {
+		// TODO: Should we enforce this if preserve_http1_header_case is set ?
```

Comment:
We shouldn't need to change the codec type; I believe the stateful formatter is safe to apply to http2 codecs (and integration tests should prove whether or not this is the case)

Reply:
Yes just came to this conclusion as well.